src/cts/src/TreeBuilder.h

```diff
@@ -50,6 +50,29 @@ namespace utl {
 class Logger;
 }  // namespace utl
 
+// Hash and compare functions for occupiedLocations_ map
+// Use this hash table to keep track of occupied locations
+namespace std {
+template <>
+struct hash<cts::Point<double>>
+{
+  std::size_t operator()(const cts::Point<double>& point) const
+  {
+    return point.getX() + (point.getY() * 11.0);
```

Comment:
This is a weak hash fn - better to use boost::hash for a pair

Reply:
OK, let me look into boost::hash.

Reply:
Replaced it with boost::unordered_set.